Sources/Fluent/Query/Query.swift

```diff
@@ -49,50 +56,82 @@ public class Query<T: Model> {
         Creates a new `Query` with the
         `Model`'s database.
     */
-    public init() {
+    public init(_ database: Database) {
         filters = []
         action = .fetch
-        database = T.database
+        self.database = database
+        unions = []
     }
 
+    var idKey: String {
+        return database.driver.idKey
+    }
+
+
     /**
         Runs the query given its properties
-        and current state. 
-     
-        Returns an array of entities.
+        and current state.
+
+        Returns the Node from the driver.
     */
-    // @discardableResult
-    func run() throws -> [T] {
-        var models: [T] = []
+    @discardableResult
+    public func raw() throws -> Node {
+        return try database.driver.query(self)
+    }
 
-        let results = try database.driver.query(self)
+    /**
+        Runs the query using Raw then converts
+        the results to an array of models.
+    */
+    @discardableResult
+    public func run() throws -> [T] {
+        var models: [T] = []
 
-        for result in results {
-            var model = T(serialized: result)
-            model.id = result[database.driver.idKey]
-            models.append(model)
+        if case .array(let array) = try raw() {
+            for result in array {
+                do {
+                    var model = try T(with: result)
+                    if case .object(let dict) = result {
+                        model.id = dict[database.driver.idKey]
+                    }
+                    models.append(model)
+                } catch {
+                    // FIXME: do what when init fails?
```

Comment:
is it possible for us to propogate up instead of catching or is that not the right behavior?